api/service/stagedstreamsync/syncing.go

```diff
@@ -285,7 +285,7 @@ func (s *StagedStreamSync) doSync(downloaderContext context.Context, initSync bo
 	}
 
 	// add consensus last mile blocks
-	if s.consensus != nil {
+	if s.consensus != nil && s.consensus.ShardID == 0 {
```

Comment:
if you need to have this only for beacon nodes, please use   `... && s.isBeaconNode` because the epoch chain shard number is also zero.